docs(security): updates content for securing kafka access (#10071)
strimzi/strimzi-kafka-operator: documentation/modules/overview/con-configuration-points-listeners.adoc

// assembly-deploy-client-access.adoc

[id="configuration-points-listeners-{context}"]
= Configuring listeners to connect to Kafka

[role="_abstract"]
Use listeners to enable client connections to Kafka.
Strimzi provides a generic `GenericKafkaListener` schema with properties to configure listeners through the `Kafka` resource.

When configuring a Kafka cluster, you specify a listener `type` based on your requirements, environment, and infrastructure. 
Services, routes, load balancers, and ingresses for clients to connect to a cluster are created according to the listener type.

Internal and external listener types are supported.

Internal listeners:: Use internal listener types to connect clients within a kubernetes cluster.
+
* `internal` to connect within the same Kubernetes cluster
* `cluster-ip` to expose Kafka using per-broker `ClusterIP` services
+
Internal listeners use a headless service and the DNS names assigned to the broker pods. 
By default, they use the Kubernetes service DNS domain (typically `.cluster.local`). 

> However, you can customize this configuration using the `useServiceDnsDomain` property. 

Reviewer: They do not use the DNS suffix by default IIRC. `useServiceDnsDomain` defaults to false.

Author: Right. According to the API ref, they don't use the DNS suffix by default.

Consider using a `cluster-ip` type listener if routing through the headless service isn't feasible or if you require a custom access mechanism, such as when integrating with specific Ingress controllers or the Kubernetes Gateway API.

External listeners:: Use external listener types to connect clients outside a Kubernetes cluster.
+
* `nodeport` to use ports on Kubernetes nodes
* `loadbalancer` to use loadbalancer services

> * `ingress` to use Kubernetes `Ingress` and the {NginxIngressController} (Kubernetes only)

Reviewer: Strictly speaking, you can install the Nginx Ingress controller for example on oPenShift as well. So I think the focus should be on the controller rather on Kubernetes.

Author: We say "(Kubernernetes only)" in a few places when referring to listener types as we don't include the ingress procedure downstream.

* `route` to use OpenShift `Route` and the default HAProxy router (OpenShift only)
+
External listeners handle access to a Kafka cluster from networks that require different authentication mechanisms.
For example, loadbalancers might not be suitable for certain infrastructure, such as bare metal, where node ports provide a better option.

> IMPORTANT: Do not use `ingress` on OpenShift, use the `route` type instead. The Ingress NGINX Controller is only intended for use on Kubernetes. The `route` type is only supported on OpenShift.

Reviewer: As above, this is not really about Kube versus OpenShift. This is about having the right Ingress controller.

```suggestion
IMPORTANT: Do not use the built in `ingress` controller on OpenShift, use the `route` type instead.
```
